napari/components/_layer_slicer.py

```diff
@@ -36,7 +37,14 @@
 # result of ``_slice_layers`` cannot be fixed to a single type.
 
 _SliceResponse = TypeVar('_SliceResponse')
-_SliceRequest = Callable[[], _SliceResponse]
+
+
+class _SliceRequest(abc.ABC, Generic[_SliceResponse]):
```

Comment:
I'd prefer this be a protocol instead of a generic because this ABC isn't actually inherited anywhere. See https://github.com/dstansby/napari/pull/3 for my suggestion for doing that keeps mypy happy and also simplifies the typing here.

Reply:
I made this update.